scripts/popup.js

```diff
@@ -1,3 +1,98 @@
+function getFormData(){
+    var formElements = document.getElementById('metadata-form').elements;
+    var formData = {};
+
+    for(var i = 0 ; i < formElements.length ; i++){
```

Comment:
Use this kind of for-loop for cleanliness: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/statements/for...in